static/js/composebox_typeahead.js

```diff
@@ -1012,10 +1021,15 @@ export function content_typeahead_selected(item, event) {
         }
     }
 
-    // Keep the cursor after the newly inserted text, as Bootstrap will call textbox.change() to
-    // overwrite the text in the textbox.
+    // Keep the cursor after the newly inserted text / selecting the placeholder text, as Bootstrap will call
+    // textbox.change() to overwrite the text in the textbox.
     setTimeout(() => {
-        textbox.caret(beginning.length, beginning.length);
+        if (item.placeholder) {
+            textbox.get(0).setSelectionRange(highlight.start, highlight.end);
+            textbox.trigger("focus");
```

Comment:
This doesn't set the caret; is that expected?  I guess one can't do slash command typeahead with a non-empty compose box anyway?

Reply:
> This doesn't set the caret; is that expected? I guess one can't do slash command typeahead with a non-empty compose box anyway?

I'm not sure I understand... As far as I know, the caret is only set when there's no selected text. But if you mean why the highlighting doesn't consider the position of the slash command, then yes, that is intentional since slash commands have to be the first thing in the compose box.